src/WebJobs.Script/Host/ScriptHostIdProvider.cs

```diff
@@ -65,7 +65,7 @@ internal static HostIdResult GetDefaultHostId(IEnvironment environment, ScriptAp
             {
                 // The hostid is derived from the hostname for Linux consumption.
                 string hostName = environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName);
-                hostId = hostName?.Replace(".azurewebsites.net", string.Empty);
+                hostId = hostName?.Replace(environment.GetDnsSuffixWithDotPrefix(), string.Empty);
```

Comment:
For existing clouds with the new handling, this will cause host IDs to change and break a number of features for those sites. Everything the host scopes, including HTTP API keys, would be invalidated.

Reply:
@balag0 - as this is a known issue with blob triggers apps on linux consumption - lets scope this to Linux Consumption ? Also, do you know if this works on Linux Dedicated Sku?

Another consideration - provide opt-in flag for Cx to ensure we do not break any existing sceanarios that are currently running in Linux Consumption - non-public cloud

Reply:
Removing the change to HostNameProvider would scope it to just Linux consumption.
It doesnt break on other skus.

@fabiocav mentioned making this change as part of vNext which I think also works since things have been broken for a while.